csharp/Facebook.Yoga/YogaNode.cs

```diff
@@ -546,13 +547,21 @@ public void SetMeasureFunction(MeasureFunction measureFunction)
         {
             _measureFunction = measureFunction;
             _ygMeasureFunc = measureFunction != null ? MeasureInternal : (YogaMeasureFunc)null;
+
+            var handle = GCHandle.Alloc(this);
+            Native.YGNodeSetContext(_ygNode, GCHandle.ToIntPtr(handle));
+
             Native.YGNodeSetMeasureFunc(_ygNode, _ygMeasureFunc);
         }
 
         public void SetBaselineFunction(BaselineFunction baselineFunction)
         {
             _baselineFunction = baselineFunction;
             _ygBaselineFunc = baselineFunction != null ? BaselineInternal : (YogaBaselineFunc)null;
+
+            var handle = GCHandle.Alloc(this);
```

Comment:
Your patch never calls `GCHandle.Free`.

Also you allocate a GCHandle both SetMeasureFunction and SetBaselineFunction, if both are called one of the values stored with YGNodeSetContext will be overwritten. The same will happen if either function is called more than once.